pandas/tests/generic/test_generic.py

```diff
@@ -232,11 +232,8 @@ def test_size_compat(self, frame_or_series):
     def test_split_compat(self, frame_or_series):
         # xref GH8846
         o = construct(frame_or_series, shape=10)
-        with tm.assert_produces_warning(
-            FutureWarning, match=".swapaxes' is deprecated", check_stacklevel=False
-        ):
-            assert len(np.array_split(o, 5)) == 5
-            assert len(np.array_split(o, 2)) == 2
+        assert len(np.array_split(o, 5)) == 5
+        assert len(np.array_split(o, 2)) == 2
```

Comment:
hi 👋 ref https://github.com/numpy/numpy/issues/24889#issuecomment-1981503361

after this PR, calling `np.array_split(df)` will no longer return `list[pd.DataFrame]`, but `list[np.ndarray]`. this was not tested here ^

could you fix the split compat? currently, this PR poses quite a nasty breaking change with no obvious migration / alternative at hand.

Reply:
cc @mroeschke @luke396 via https://github.com/pandas-dev/pandas/pull/51960#discussion_r1142438800